### ISSUE TITLE
Automatically open Full Description (F3)

### ISSUE THREAD

**Issue by luapfr** [sabrogden/Ditto#507]
How could I configure it to always automatically open the "Full Complete Description" without having to press F3.

Look in the settings but I can't find it.

**Comment by luapfr** [comment]
Can someone help me ?

It is possible ?

**Comment by luapfr** [comment]
If this is not possible, could you add a feature for a new version ?

**Comment by sabrogden** [comment]
You can always show the description window by double clicking on the title bar.  This will set it to be always on top.

**Comment by luapfr** [comment]
Here it doesn't work
Whenever I open the main screen I need to press F3

But I need to close the Full Description screen, I just want it to appear automatically when I open Ditto

See the attached video.
https://mega.nz/file/0vxUmIbB#CDP_4EiFEhltwAM-Q5PcpoQCswE1iYcRxf9EFM7saS8

**Comment by luapfr** [comment]
Hello [sabrogden](https://github.com/sabrogden)
?

**Comment by Chealer** [comment]
@luapfr: Which description do you want to open? Ticket #172 requests to show the description when hovering.

**Comment by luapfr** [comment]
I don't want to hover the mouse over it and see a hint; I want a setting where the full description screen automatically opens when I open Ditto.

https://mega.nz/file/0vxUmIbB#CDP_4EiFEhltwAM-Q5PcpoQCswE1iYcRxf9EFM7saS8

**Comment by Chealer** [comment]
@luapfr: Again, **which** description? You want the full description of all entries to be automatically shown?

**Comment by luapfr** [comment]
> [@luapfr](https://github.com/luapfr): Again, **which** description? You want the full description of all entries to be automatically shown?

Yes

**Comment by Chealer** [comment]
Thank you @luapfr 

The CopyQ manager displays the full contents of each clip in its main window (by default). Is that the behavior you want? Or do you also want all of Ditto's full description window to be shown (including clip properties, and length statistics [the number of characters and words])?

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by luapfr** [comment]
Yes, copyQ has a very interesting layout where it shows the list of saved items on one side and a preview of the items on the other.

I believe the same result can be achieved with ditto if F3 is automatically activated when navigating through the items. 



**The CopyQ manager displays the full contents of each clip in its main window (by default). Is that the behavior you want**  ?  Yes

**Comment by Chealer** [comment]
> Yes, copyQ has a very interesting layout where it shows the list of saved items on one side and a preview of the items on the other.

Really? Can you provide a screenshot showing that?

I see 2 solutions to your issue:
1. a _setting_ which causes all clips to always display their full contents
2. and/or some action (perhaps a keyboard shortcut) which would toggle between displaying all clips truncated to/from displaying them fully expanded.

Can you confirm that you would prefer #1 to #2?

**Comment by luapfr** [comment]
It could be a setting that, when activated upon opening the main screen and scrolling through items with the keyboard arrows or mouse click, would automatically trigger the preview screen without needing to press F3.


Pode ser uma configuração que quando ativa ao abrir a tela principal e rolar pelos itens com seta do teclado ou click do mouse já acionaria a tela de preview de forma automática sem precisar teclar F3

**Comment by Chealer** [comment]
Thanks @luapfr 

I assume the main reason why the tooltip does not satisfy you is that it is truncated?

Would an implementation of what ticket #172 requests fully solve your issue if it was done only when each clip was selected (rather than merely hovered as requested)?

**Comment by luapfr** [comment]
> Thanks [@luapfr](https://github.com/luapfr)
> 
> I assume the main reason why the tooltip does not satisfy you is that it is truncated?
> 
> Would an implementation of what ticket [#172](https://github.com/sabrogden/Ditto/issues/172) requests fully solve your issue if it was done only when each clip was selected (rather than merely hovered as requested)?


The Hint is not a good solution; the Description Screen (F3) is better because it allows me to select something within a long description, which is not possible with the hint.

a Dica ( Hint ) Não é uma solução boa, o melhor a Tela de Descrição ( F3 ) pois com essa tela eu consigo selecionar algo dentro de uma descrição grande, a dica isso não é possível

**Comment by Chealer** [comment]
Thanks

Would it fully solve your issue if Ditto kept displaying a tooltip when a clip is hovered, but replaced it with the full description window when the clip was selected?

If not, please indicate the version you use and describe the issue you are experiencing.

**Comment by luapfr** [comment]
I'm using the version 3_25_113_0

The Hint isn't a viable solution.

The best option is the Description screen automatically activated by F3, as I mentioned.

The Hint is bad; for example, if I have a large text and want to select a word, I can't do it with a Hint, but it's possible with the Description screen.

**Comment by Chealer** [comment]
Thanks

> The Hint isn't a viable solution.
> 
> The best option is the Description screen automatically activated by F3, as I mentioned.
> 
> The Hint is bad; for example, if I have a large text and want to select a word, I can't do it with a Hint, but it's possible with the Description screen.

Can you clarify your point? Are you asking to remove (or fully replace) the currently shown tooltips?

**Comment by luapfr** [comment]
Tips may exist, but they are not viable for displaying the complete content of a history, as I said before, it's not possible to select a word or part of the content within a tip.

**Comment by Chealer** [comment]
Right, but if you are not against keeping tooltips, I fail to understand your previous comment. Again, would it fully solve your issue if Ditto kept displaying a tooltip when a clip is hovered, but replaced it with the full description window when the clip is selected? And if not, what exactly do you want?

**Comment by luapfr** [comment]
I don't know how I could explain it better.

I believe there could be a setting where F3 is automatically activated when I click with the mouse, type something on the keyboard, or scroll using the arrow keys, so the display screen would be shown for those who like it. However, those who want hints could choose not to activate this setting.

With this setting, Ditto would look similar to CopyQ, always displaying the full description.


Não sei como poderia te explicar melhor.

Acredito que poderia ter uma configuração onde o F3 fosse acionado automaticamente quando eu clicar com o mouse ou digitar algo no teclado ou rolar pelas setas assim a tela de exibição seria mostrado para que gosta dela porém quem quiser dica Hint poderia não ativar essa configuração.
 Com essa configuração o ditto ficaria parecido com o CopyQ sempre exibindo a descrição completa.

**Comment by Chealer** [comment]
Thanks for your effort @luapfr. Unfortunately I hardly understand Portuguese, and I do not think your English explanation alone suffices to implement your suggestion. Although you are open to make the behavior configurable, this would represent a considerable change to Ditto, so it needs to be clear what you are proposing and what would be the impacts.

> I believe there could be a setting where F3 is automatically activated when I click with the mouse, type something on the keyboard, 

This is too vague. By "type something", do you mean something specific, like a specific key, or what? You need to make it clear in which conditions a clip's full description would be shown (or hidden), and ideally in which conditions the tooltip would keep showing.

> With this setting, Ditto would look similar to CopyQ, always displaying the full description.

I am sorry, but I am getting even more confused. CopyQ always displays the full contents of all clips. This is not at all like automatically opening Ditto's Full Description, which is a window different from Ditto's main window.

You have written more than 10 comments in this ticket. Can you please edit the ticket body to recapitulate what was said? Ideally, this would make it clear which issue you report, and which solution(s) you are proposing.

**Comment by luapfr** [comment]
I'll explain it to you again.

When I open the Ditto main screen, I need to press F3 to show the entire item description.

Instead of having to press F3 automatically when I click on an item or scroll with the arrow keys, I'd like the item description to automatically open.

What I'm asking for, in short, is a setting that always shows the complete description screen. I don't want a hint; I want the full description screen.


**Ditto already has a full description screen that could somehow always be displayed without me having to keep pressing F3.**

**Comment by garoto** [comment]
> I'll explain it to you again.

Waste of time. The person you're engaging with is not a member, regular contributor nor a C++ programmer familiar with the WinAPIs ditto makes use of. Meaning, he won't be making a PR at the end of all of this addressing your needs. His whole shtick is to suck his interlocutor's sanity dry by going on an endless back-and-forth of ever increasing minutia, all the while blaming (in a passive-aggressive way) the other party of being "unclear", "incoherent", "too general" or any variations of the term.

Just input in the issues tab of this repo the query "is:issue commenter:Chealer" and see for yourself.
